frontend/src/App.tsx

```diff
@@ -19,17 +19,42 @@ import { useEffect, useState } from 'react'
 import { User, onAuthStateChanged } from 'firebase/auth'
 import { AuthProvider, PrivateRoute, PublicRoute } from '@auth'
 import { auth } from '@fire'
+import axios from 'axios'
 
 const App = () => {
   const [currentUser, setCurrentUser] = useState<User | null>(null)
   const [isLoading, setIsLoading] = useState(true)
+  const [token, setToken] = useState('')
 
   useEffect(() => {
     onAuthStateChanged(auth, (user) => {
       setCurrentUser(user)
-      setIsLoading(false)
+      // need these to conditions to resolve isLoading at the correct time so data can be loaded properly
+      if (user)
+        user
+          .getIdToken(false)
+          .then((idToken) => {
+            // store token in local storage
+            axios.interceptors.request.use(
+              (request) => {
+                request.headers.authtoken = idToken
+                return request
+              },
+              (error) => {
+                return Promise.reject(error)
+              }
+            )
+            setToken(idToken) // might have to leave this in here
```

Comment:
Wait why do we need this, I tried removing it and the useState for token and it still appears to work fine for me?

Reply:
I guess this is one of the ones not needed lol